Transaction manager maintenance doc

poornachandra opened the pull request:
Sending out the PR early to get feedback for Operations section. Still need to document the Transaction Manager REST APIs.

anew commented on cdap-docs/admin-manual/source/operations/tx-maintenance.rst:
```diff
@@ -0,0 +1,23 @@
+.. highlight:: console
+
+Transactions are marked as invalid when the changes made during the transaction cannot be rolled back. Transaction
+Manager has to store all invalid transactions so that they can be excluded from all future reads. This invalid 
```
I am wondering if we should call it Transaction Service? That is what it is called in the UI: "transaction 
Service that maintains transaction states."

anew replied:
How about: 

The Transaction Service keeps track of all invalid transactions to exclude their writes from all future reads. Over time this invalid list can grow and lead to performance degradation. To avoid that, you can prune the invalid list after the data of invalid transactions has been removed---which happens during major HBase compactions. 

After a major...

ghelmling commented on cdap-docs/admin-manual/source/operations/tx-maintenance.rst:
```diff
@@ -0,0 +1,23 @@
+to performance degradation.
+
+Since data written during invalid transaction gets removed during major compaction, invalid transaction list can be 
+manually pruned after a major compaction in HBase has completed.
```
"after a major compaction on all transactional tables in HBase..."

poornachandra commented:
@anew @ghelmling @awholegunch Addressed review comments, please take a look.

ParthGandhi commented:
@poornachandra Remember to add links to this page from the REST APIs as well: https://github.com/caskdata/cdap/pull/1630

awholegunch commented on cdap-docs/admin-manual/source/operations/tx-maintenance.rst:
```diff
@@ -0,0 +1,19 @@
+    :author: Cask Data, Inc.
+    :copyright: Copyright © 2015 Cask Data, Inc.
+
+===============================
```
add ref above the over line:
```
.. _tx-maintenance:
```

awholegunch commented on cdap-docs/admin-manual/source/operations/tx-maintenance.rst:
```diff
@@ -0,0 +1,19 @@
+.. highlight:: console
+
+The Transaction Service keeps track of all invalid transactions to exclude their writes from all future reads. 
+Over time this invalid list can grow and lead to performance degradation. To avoid that, you can prune the invalid 
```
```
Over time, this *invalid* list can grow and lead to performance degradation. 
To avoid a performance hit, you can prune the *invalid * list after the data of the invalid transactions 
has been removed |---| which happens during major HBase compactions of the 
transactional tables.
```

awholegunch commented on cdap-docs/admin-manual/source/operations/tx-maintenance.rst:
```diff
@@ -0,0 +1,19 @@
+After a major compaction that started at time ``n`` completes, the invalid transaction list can be safely pruned
+until ``n - 1 day``, using the REST call at [Insert ref to REST API]
+
+The current length of invalid transaction list can be obtained using the REST call [Insert ref to REST API]
```
Insert a blank line at the end of the file.

awholegunch replied:
```
The current length of the invalid transaction list can be obtained 
using the :ref:`Transaction Service HTTP RESTful API <http-restful-api-transactions>`.

```

awholegunch commented on cdap-docs/admin-manual/source/operations/tx-maintenance.rst:
```diff
@@ -0,0 +1,19 @@
+transactional tables.
+
+After a major compaction that started at time ``n`` completes, the invalid transaction list can be safely pruned
+until ``n - 1 day``, using the REST call at [Insert ref to REST API]
```
```
After a major compaction that started at time ``n`` completes, the invalid transaction list can be safely pruned
until ``n -1 day``, using the :ref:`Transaction Service HTTP RESTful API <http-restful-api-transactions>`.
```
Given that the API is milliseconds, what is `n -1 day`? Should it be `n -86400000`? Or does `day` get converted?

awholegunch replied:
As anew suggested, could we make this ``t -1 day``?

anew commented on cdap-docs/admin-manual/source/operations/tx-maintenance.rst:
```diff
@@ -0,0 +1,19 @@
+list after the data of invalid transactions has been removed---which happens during major HBase compactions of the 
+transactional tables.
+
+After a major compaction that started at time ``n`` completes, the invalid transaction list can be safely pruned
```
can we call n t?

poornachandra commented:
@awholegunch Addressed review comments.

Right now, in tx-maintenance.rst we refer to two REST APIs using the same reference. Can we instead have individual references for the APIs to make things clear in the doc?

poornachandra commented:
Thanks @awholegunch. LGTM

awholegunch commented:
LGTM.
